File: frontend/src/features/settings/SettingsDTO.ts
import {Settings} from "./Settings";
import {Email} from "../users/Email";

export interface SettingsDTO {
    email: string;
    soundVolume: number;
    musicVolume: number;
}

export const toDomainObject = (dto: SettingsDTO): Settings => ({
    email: Email.of(dto.email),
    soundVolume: dto.soundVolume,
    musicVolume: dto.musicVolume
});

File: mobile/app/src/main/java/com/bellako/kiwi/features/settings/data/SettingsDTO.kt
package com.bellako.kiwi.features.settings.data

import com.bellako.kiwi.features.users.data.Email

data class SettingsDTO(
    val email: String,
    val soundVolume: Float,
    val musicVolume: Float,
) {
    fun toState(): SettingsState =
        SettingsState(
            email = email,
            soundVolume = soundVolume,
            musicVolume = musicVolume,
        )

    fun toDomainObject(): Result<Settings> =
        Email.of(email).map { validEmail ->
            Settings(
                email = validEmail,
                soundVolume = soundVolume,
                musicVolume = musicVolume,
            )
        }
}
